File: src/php/Linpack/LinpackBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Linpack {

    use EntityFX\NetBenchmark\Core\Writer;
    use EntityFX\NetBenchmark\Core\Generic\BenchmarkBase;

    class LinpackBenchmark extends BenchmarkBase {

        private $linpack;

        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Ratio = 10;
            $this->linpack = new Linpack($printToConsole);
        }

        public function Warmup($aspect = 0.05) {
        }

        protected function BenchImplementation() {
            return $this->linpack->run_benchmark(2000);
        }

        protected function PopulateResult($benchResult, $result) {
            $benchResult["Points"] = $result["MFLOPS"] * $this->Ratio;
            $benchResult["Result"] = $result["MFLOPS"];
            $benchResult["Units"] = "MFLOPS";
            $benchResult["Output"] = $result["Output"];
            return $benchResult;
        }
    }
}

File: src/lua/linpack/linpackBenchmark.lua
require "linpack/linpack"

LinpackBenchmark = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.name = "LinpackBenchmark"
    a.ratio = 10
    a.linpack = Linpack(printToConsole)
end)

function LinpackBenchmark:benchImplementation()
    return self.linpack:run_benchmark(1000)
end

function LinpackBenchmark:warmup(aspect)

end

function LinpackBenchmark:populateResult(benchResult, result)
    benchResult.points = result.mflops * self.ratio
    benchResult.result = result.mflops
    benchResult.units = "MFLOPS"
    benchResult.output = result.Output
    return benchResult
end